grovel/asdf.lisp

```diff
@@ -135,19 +139,37 @@
         (multiple-value-bind (files translatedp) (call-next-method)
           (values (append files (list lib-file o-file)) translatedp)))))
 
+(defun absolute-cache-dir (input-file cache-dir)
+  (when cache-dir
+    (let ((input-dir (uiop:pathname-directory-pathname input-file)))
+      (uiop:ensure-directory-pathname
+       (uiop:subpathname input-dir cache-dir)))))
+
+(defun system-to-component-path (component)
+  (let* ((sys (component-system component))
+         (sys-dir (pathname-directory (component-pathname sys)))
+         (comp-dir (pathname-directory (component-pathname component))))
+    (make-pathname* :directory (cons :relative (subseq comp-dir (length sys-dir))))))
+
```

Comment:
Given an asdf component, this function returns the relative path from the system to that component